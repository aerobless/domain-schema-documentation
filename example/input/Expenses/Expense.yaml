$id: /Expenses/Expense.yaml
title: Expense
description: An expense
type: object
x-schema-type: Aggregate
examples:
  - id: "exp123"
    date: "2023-10-01"
    amount: 200.50
    currency: "USD"
    description: "Office Supplies"
    userId: "user456"
    costCenter: "CC789"
    status: "Submitted"
    journal:
      - id: "jrn1"
        date: "2023-10-01"
        newStatus: "Submitted"
        userId: "user456"
        comment: "Initial submission"
properties:
  id:
    type: string
    description: The unique identifier of the expense
  date:
    type: string
    format: date
    description: The date of the expense
  amount:
    type: number
    format: float
    description: The amount of the expense
  currency:
    type: string
    description: The currency of the expense
  description:
    type: string
    description: The description of the expense
  userId:
    type: string
    x-references: ../Users/User.yaml
    description: The user who created the expense
  costCenter:
    type: string
    description: The cost center of the expense
    x-references: ../Users/Project.yaml
  status:
    $ref: ./ExpenseStatus.yaml
  journal:
    type: array
    description: The journal of the expense
    items:
      $ref: ./ExpenseJournal.yaml
