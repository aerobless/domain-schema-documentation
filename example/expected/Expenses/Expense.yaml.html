<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.84.0">
    <title>Test Application</title>
    <link rel="canonical" href="https://getbootstrap.com/docs/5.0/examples/navbar-static/">
    <link rel=stylesheet href=https://cdn.jsdelivr.net/npm/pretty-print-json@3.0/dist/css/pretty-print-json.css>
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet">
    <meta name="theme-color" content="#7952b3">
    <style>
    .alert ul {
      margin-bottom: 0;
    }
    .mermaid{
      height: 300px;
      border: 1px solid #000;
    }
    .mermaid pre {
      display: none;
    }
    section {
      margin-bottom: 50px;
    }
    .title-badge {
      font-size: 0.3em;
      vertical-align: top;
    }
    .json {
      border: 1px solid black;
      background-color: lightgray;
      padding: 10px;
     }
    </style>
  </head>
  <body>
    <main class="container">
      <nav aria-label="breadcrumb">
        <ol class="breadcrumb">
          <li class="breadcrumb-item"><a href="./../../index.html">Application</a></li>
          <li class="breadcrumb-item"><a href="./index.html">Expenses</a></li>
          <li class="breadcrumb-item active" aria-current="page">Expense</li>
        </ol>
      </nav>
      
      <h1>Expense</h1>
      
      
      <section>
        <p>An expense</p>
        <div class="mermaid">
          <pre class="mermaidDefinition">
            classDiagram
      namespace Expenses {
        class _Expenses_Expense_yaml["Expense"]
        class _Expenses_ExpenseStatus_yaml["Expense Status"]
        class _Expenses_ExpenseJournal_yaml["Expense Journal Entry"]
      }
      namespace Users {
        class _Users_User_yaml["User"]
        class _Users_Project_yaml["Project"]
      }
      _Expenses_Expense_yaml ..> _Users_User_yaml :userId
      _Expenses_Expense_yaml ..> _Users_Project_yaml :costCenter
      _Expenses_Expense_yaml o--" N" _Expenses_ExpenseJournal_yaml :journal
      click _Expenses_Expense_yaml href "./Expense.yaml.html" "Expense"
      click _Users_User_yaml href "./../Users/User.yaml.html" "User"
      click _Users_Project_yaml href "./../Users/Project.yaml.html" "Project"
      click _Expenses_ExpenseStatus_yaml href "./ExpenseStatus.yaml.html" "Expense Status"
      click _Expenses_ExpenseJournal_yaml href "./ExpenseJournal.yaml.html" "Expense Journal Entry"
          </pre>
        </div>
      </section>
      
      
      
      <section>
        <h2>Properties</h2>
        <table class="table">
          <thead>
            <tr>
              <th>Name</th>
              <th>Type</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>id</td>
              <td>string</td>
              <td>The unique identifier of the expense</td>
            </tr>
            <tr>
              <td>date</td>
              <td>date</td>
              <td>The date of the expense</td>
            </tr>
            <tr>
              <td>amount</td>
              <td>float</td>
              <td>The amount of the expense</td>
            </tr>
            <tr>
              <td>currency</td>
              <td>string</td>
              <td>The currency of the expense</td>
            </tr>
            <tr>
              <td>description</td>
              <td>string</td>
              <td>The description of the expense</td>
            </tr>
            <tr>
              <td>userId</td>
              <td>References <a href="./../Users/User.yaml.html">User</a></td>
              <td>The user who created the expense</td>
            </tr>
            <tr>
              <td>costCenter</td>
              <td>References <a href="./../Users/Project.yaml.html">Project</a></td>
              <td>The cost center of the expense</td>
            </tr>
            <tr>
              <td>status</td>
              <td><a href="./ExpenseStatus.yaml.html">Expense Status</a></td>
              <td></td>
            </tr>
            <tr>
              <td>journal</td>
              <td>[<a href="./ExpenseJournal.yaml.html">Expense Journal Entry</a>]</td>
              <td>The journal of the expense</td>
            </tr>
          </tbody>
        </table>
      </section>
      
      <section>
        <h2>Examples</h2>
        <pre class="json">
        {"id":"exp123","date":"2023-10-01","amount":200.5,"currency":"USD","description":"Office Supplies","userId":"user456","costCenter":"CC789","status":"Submitted","journal":[{"id":"jrn1","date":"2023-10-01","newStatus":"Submitted","userId":"user456","comment":"Initial submission"}]}
        </pre>
      </section>
      
      <section>
        <h2>Links</h2>
        <ul>
          <li><a href="./java/Expense.java">Java-File</a></li>
        </ul>
      </section>
      
      
      
      
    </main>
    <script src="https://cdn.jsdelivr.net/npm/svg-pan-zoom@3.5.0/dist/svg-pan-zoom.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/pretty-print-json@3.0/dist/pretty-print-json.min.js"></script>
    <script type="module">
      import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs';

      function prettifyJson(){
        const elements = document.querySelectorAll('.json');
        for(const element of elements) {
          element.innerHTML = prettyPrintJson.toHtml(JSON.parse(element.innerText));
        }
      }

      function drawDiagrams() {
        const elements = document.querySelectorAll('.mermaid');
        elements.forEach((element, i) =>drawDiagram(element, i));
      }

      async function drawDiagram(element, i) {
        const graphDefinition = element.querySelector(".mermaidDefinition")
        const {svg} = await mermaid.render('mySvg_' + i, graphDefinition.innerText);
        const preview = createMermaidPreview(element, svg);
        const modal = createMermaidModal();
        //Do this with a slight delay to take link in mermaid diagram take precedence
        element.addEventListener('click', () => setTimeout(() => showMermaidModal(preview,modal), 30));
      }

      function createMermaidPreview(element, svg) {
        const preview = document.createElement('div');
        preview.style.height = '100%';
        preview.style.width = '100%';
        preview.innerHTML = svg.replace(/[ ]*max-width:[ 0-9\.]*px;/i , '');
        preview.querySelector("svg").style.height = '100%';
        preview.querySelector("svg").style.width = '100%';
        element.appendChild(preview);
        return preview;
      }

      function createMermaidModal() {
        const modal = document.createElement('div');
        modal.innerHTML =
            '<div class="modal" tabindex="-1">\n' +
            '  <div class="modal-dialog modal-fullscreen">\n' +
            '    <div class="modal-content">\n' +
            '      <div class="modal-header">\n' +
            '        <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>\n' +
            '      </div>\n' +
            '      <div class="modal-body"></div>\n' +
            '    </div>\n' +
            '  </div>\n' +
            '</div>';
        document.body.appendChild(modal);
        return modal;
      }

      function showMermaidModal(preview, modal) {
        const modalBody = modal.querySelector('.modal-body');
        modalBody.innerHTML = preview.innerHTML
        const modalDiv = modal.querySelector('.modal');
        const svg = modalBody.querySelector('svg');
        new bootstrap.Modal(modalDiv).show();
        svgPanZoom(svg, {
          zoomEnabled: true,
          controlIconsEnabled: true,
          fit: true,
          center: true
        })
      }

      mermaid.initialize({ startOnLoad: false,securityLevel: 'loose'});
      drawDiagrams();
      prettifyJson();
    </script>
  </body>
</html>
